tests/Fixer/Phpdoc/PhpdocTypesOrderFixerTest.php

```diff
@@ -0,0 +1,409 @@
+                '<?php /** @method Foo|\Bar foo(Foo|\Bar $foo, Foo|\Bar $bar) */',
+            ],
+            [
+                '<?php /** @var null|array|Bar\Baz|bool[]|false|Foo|int|object|resource|string|string[] */',
+                '<?php /** @var string[]|resource|false|object|null|Foo|Bar\Baz|bool[]|string|array|int */',
+            ],
+            [
+                '<?php /** @var null|array<int, string> Foo */',
+            ],
+            [
+                '<?php /** @var null|array<int, array<string>> Foo */',
+            ],
+            [
+                '<?php /** @var NULL|int|string */',
+                '<?php /** @var string|int|NULL */',
+            ],
+            [
+                '<?php /** @var ?Bar|Foo */',
+                '<?php /** @var Foo|?Bar */',
+            ],
+            [
+                '<?php /** @var Bar|?Foo */',
+                '<?php /** @var ?Foo|Bar */',
+            ],
+            [
+                '<?php /** @var ?\Bar|Foo */',
+                '<?php /** @var Foo|?\Bar */',
+            ],
+            [
+                '<?php /** @var Bar|?\Foo */',
+                '<?php /** @var ?\Foo|Bar */',
+            ],
+        ];
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider getFixWithAlphaAlgorithmAndNullAlwaysLastCases
+     */
+    public function testFixWithAlphaAlgorithmAndNullAlwaysLast($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'sort_algorithm' => 'alpha',
+            'null_position' => 'always_last',
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    public function getFixWithAlphaAlgorithmAndNullAlwaysLastCases()
+    {
+        return [
+            [
+                '<?php /** @var int|string|null */',
+                '<?php /** @var string|int|null */',
+            ],
+            [
+                '<?php /** @param Bar|\Foo */',
+                '<?php /** @param \Foo|Bar */',
+            ],
+            [
+                '<?php /** @property-read \Bar|Foo */',
+                '<?php /** @property-read Foo|\Bar */',
+            ],
+            [
+                '<?php /** @property-write bar|Foo */',
+                '<?php /** @property-write Foo|bar */',
+            ],
+            [
+                '<?php /** @return Bar|foo */',
+                '<?php /** @return foo|Bar */',
+            ],
+            [
+                '<?php /** @method \Bar|Foo foo(\Bar|Foo $foo, \Bar|Foo $bar) */',
+                '<?php /** @method Foo|\Bar foo(Foo|\Bar $foo, Foo|\Bar $bar) */',
+            ],
+            [
+                '<?php /** @var array|Bar\Baz|bool[]|false|Foo|int|object|resource|string|string[]|null */',
+                '<?php /** @var string[]|resource|false|object|null|Foo|Bar\Baz|bool[]|string|array|int */',
+            ],
+            [
+                '<?php /** @var array<int, string>|null Foo */',
```

Comment:
There is one below this one. Did you mean to sort collection key/value types as well, e.g.:
```diff
-/** @var array<Foo|null> */
+/** @var array<null|Foo> */
```